### PR TITLE
Prevent describe blocks with the same name in ExUnit

### LINKED ISSUE #6236: Confusion with multiple describe blocks with the same message
After getting to the bottom of what seemed like a weird test failure, I realized I made a mistake by giving two `describe` blocks the same description.

Along with setup blocks, this causes a confusion: the second `setup` is executed for the first test, not the first `setup` block as I'd expect to.

### Current behavior

```elixir
defmodule DescribeTest do
  use ExUnit.Case

  describe "my_description" do
    setup do
      {:ok, %{value: 5}}
    end

    test "test 1", %{value: value} do
      assert 5 = value
    end
  end

  describe "my_description" do
    setup do
      {:ok, %{value: 10}}
    end

    test "test 2", %{value: value} do
      assert 10 = value
    end
  end
end
```

Code above will fail with the following error:

```
mix test test/unit/describe_test.exs


  1) test my_description test 1 (DescribeTest)
     test/unit/describe_test.exs:9
     match (=) failed
     code:  5 = value
     right: 10
     stacktrace:
       test/unit/describe_test.exs:10: (test)

.

Finished in 0.08 seconds
2 tests, 1 failure

Randomized with seed 873389
```

### Expected behavior

Show a warning or even raise an error when defining a `describe` block with description that was already taken. I'm not sure for a use case to justify having multiple `describe` blocks with the same description.

### PR DESCRIPTION
**Pull request by whatyouhide**
Closes #6236.

### CODE REVIEW

File: lib/ex_unit/lib/ex_unit/case.ex [hunk 1]
```diff
@@ -356,10 +360,20 @@ defmodule ExUnit.Case do
               "for describe/2 on named setups and how to handle hierarchies"
       end
 
-      @ex_unit_describe (case unquote(message) do
-        msg when is_binary(msg) -> msg
-        msg -> raise ArgumentError, "describe name must be a string, got: #{inspect msg}"
-      end)
+      message =
+        case unquote(message) do
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/case.ex]: cond maybe?

```
message = unquote(message)
cond do
  not is_binary(...) ->
    ...
  message not in @ex_... ->
    ...
  true ->
    :ok
end
```

?

Comment (whatyouhide) [inline lib/ex_unit/lib/ex_unit/case.ex]: Yep good point! Updated.

**Comment by stevedomin** [comment]
Would it be worth adding this to the Elixir 1.5 changelog?

**Comment by josevalim** [comment]
Yes, please do send a PR to the v1.5 branch?

**Pull request merged**